### PR TITLE
feat(build): add gradle tomcat start with jpda property

### PR DESCRIPTION
**Pull request by ChristianMurphy**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]

##### Description of change

This gives a quick way to start Tomcat with JPDA remote debugging enabled on port 8000.

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

**Comment by drewwills** [comment]
Do we need a separate Gradle task for this capability?  I suspect it's something we could bake into the original `tomcatStart` task based on a property.

**Comment by ChristianMurphy** [comment]
We could convert `tomcatStart` to use `catalina start`, instead of the `startup` script. My reasoning against that is, if it is put behind a property, people would not be able to discover debug mode from `gradle task`

**Comment by jonathanmtran** [comment]
Instead of a `tomcatStartWithDebugger` task, what is everyone's thought on a task that serves as a wrapper for `catalina.sh` and passes any arguments along to it? 

I can see this being useful for `jpda start` as originally intended or even simply `start` for those who don't want it to wait until Tomcat's up. I can also see being able to call `version` to determine what version of Tomcat one is running. Just a thought.

**Comment by drewwills** [comment]
It's a good thought @jonathanmtran.  Rather than a (confusing) collection of tasks that start Tomcat in a variety of ways, I'd like to see a single, flexible task... together with some easy-to-find, easy-to-understand documentation for how to leverage that flexibility.

**Comment by ChristianMurphy** [comment]
Updated so that:

```sh
./gradlew tomcatStart -Pwith-jdba
```

enables the debugger

**Comment by bjagg** [comment]
:+1:

File: gradle/tasks/tomcat.gradle [hunk 1]
```diff
@@ -106,12 +106,16 @@ task tomcatStart() {
...
         logger.lifecycle("Starting Tomcat servlet container in ${serverHome}")
-        String executable = isWindows ? 'cmd' : './startup.sh'
+        String executable = isWindows ? 'cmd' : './catalina.sh'
         ant.exec(dir: "${serverHome}/bin", executable: executable, spawn: true) {
             if (isWindows) {
                 arg(value: '/c')
-                arg(value: 'startup.bat')
+                arg(value: 'catalina.bat')
             }
+            if (project.hasProperty('with-jdba')) {
```
Comment (jonathanmtran) [inline gradle/tasks/tomcat.gradle]: Shouldn't `with-jdba` be `with-jpda` ?

Comment (ChristianMurphy) [reply gradle/tasks/tomcat.gradle]: yes :sweat_smile:

Comment (ChristianMurphy) [reply gradle/tasks/tomcat.gradle]: Updated

**Comment by jgribonvald** [comment]
+1

**Pull request merged**